### PR TITLE
[WIP] Stick the jwt and oidc info into kube config

### PR DESCRIPTION
**Pull request by adamcarlton**

### CODE REVIEW

File: kubeclient.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	log.Print("You are now logged in. You may close the browser window")
+	os.Exit(0)
+}
+
+func configureFile(jwt string) {
+	authProv := "--auth-provider=oidc"
+	authIssuer := fmt.Sprintf("--auth-provider-arg=idp-issuer-url=%s", issuerFlag)
+	authCliID := "--auth-provider-arg=client-id=" + os.Getenv("CLIENT_ID")
+	authCliSecret := "--auth-provider-arg=client-secret=" + os.Getenv("CLIENT_SECRET")
```
Comment (dhawal55) [inline kubeclient.go]: Why are we setting auth-provider fields here? kubectl is not talking to auth provider so we should just set token in kubectl config

File: kubeclient.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+func createMux() *http.ServeMux {
+	newMux := http.NewServeMux()
+	newMux.HandleFunc("/", localHandler)
+	return newMux
+}
+
+func parseFlags() {
+	flag.StringVar(&hostFlag, "host", os.Getenv("SERVER_HOSTNAME"), "host name to use when generating the auth url")
+	flag.StringVar(&claimsFlag, "claims", os.Getenv("JWT_CLAIMS"), "claims to use when setting kube config")
+	flag.StringVar(&issuerFlag, "issuer", os.Getenv("AUTH_ISSUER"), "auth issuer to use when setting kube config")
```
Comment (dhawal55) [inline kubeclient.go]: See above comment. JWT_CLAIMS & AUTH_ISSUER may not be needed

File: kubeclient.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func main() {
+	parseFlags()
+	authURL, portNum, err := generateAuthURL()
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	log.Print("\nFollow this url if you want to live!: ", authURL)
+	http.ListenAndServe(":"+portNum, createMux())
```
Comment (dhawal55) [inline kubeclient.go]: `http.ListenAndServe` can throw err so log it - `log.Fatal(http.ListenAndServe(...))`

Comment (adamcarlton) [inline kubeclient.go]: fixed

File: kubeclient.go (hunk 2 above)
Comment (adamcarlton) [inline kubeclient.go]: claims is not needed however issuer is

File: kubeclient.go (hunk 1 above)
Comment (adamcarlton) [inline kubeclient.go]: after testing we need provider, issuer, and client id

Comment (adamcarlton) [inline kubeclient.go]: ignore my last comment. you're totally right

**Pull request merged**